<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>869</width>
    <height>520</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Leslie's OS Course Design - U201714863</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="mem_basic">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>811</width>
      <height>421</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="PROC_info">
     <attribute name="title">
      <string>进程信息</string>
     </attribute>
     <widget class="QTableWidget" name="table1">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>461</width>
        <height>371</height>
       </rect>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>false</bool>
      </property>
      <property name="showGrid">
       <bool>false</bool>
      </property>
      <column>
       <property name="text">
        <string>进程名</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>进程PID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>进程状态</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>占用内存</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>优先级</string>
       </property>
      </column>
     </widget>
     <widget class="QGroupBox" name="proc_info">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>160</y>
        <width>291</width>
        <height>221</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>进程信息</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>60</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>父进程 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_8">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>进程组 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>用户 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>进程 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>150</y>
         <width>91</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>占用CPU(%) :</string>
       </property>
      </widget>
      <widget class="QLabel" name="ppid">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>60</y>
         <width>161</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="proc_group">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>90</y>
         <width>161</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="user">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>120</y>
         <width>161</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="pid">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>30</y>
         <width>161</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="proc_cpu">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>150</y>
         <width>161</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QPushButton" name="kill">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>180</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Kill Process</string>
       </property>
       <property name="autoExclusive">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="refresh">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>180</y>
         <width>101</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Refresh Info</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>20</y>
        <width>291</width>
        <height>131</height>
       </rect>
      </property>
      <property name="title">
       <string>概况</string>
      </property>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>30</y>
         <width>81</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>总进程数 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_14">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>60</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>运行进程 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_15">
       <property name="geometry">
        <rect>
         <x>50</x>
         <y>90</y>
         <width>71</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>睡眠进程 :</string>
       </property>
      </widget>
      <widget class="QLabel" name="total">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>30</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="running">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>60</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="sleeping">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>90</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="CPU_info">
     <attribute name="title">
      <string>CPU/内存信息</string>
     </attribute>
     <widget class="QGroupBox" name="cpu_load">
      <property name="geometry">
       <rect>
        <x>520</x>
        <y>20</y>
        <width>231</width>
        <height>171</height>
       </rect>
      </property>
      <property name="title">
       <string>CPU 负载</string>
      </property>
      <widget class="QProgressBar" name="cpu_usage">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>520</x>
        <y>220</y>
        <width>231</width>
        <height>121</height>
       </rect>
      </property>
      <property name="title">
       <string>CPU 进程/线程数</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>进程数</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>线程数</string>
       </property>
      </widget>
      <widget class="QLabel" name="proc_count">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>40</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="thread_count">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>80</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="Line" name="line">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>171</width>
         <height>16</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>461</width>
        <height>171</height>
       </rect>
      </property>
      <property name="title">
       <string>CPU 基本信息</string>
      </property>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>40</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Model Name :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>70</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>CPU MHz :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>100</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>CPU Cores :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>130</y>
         <width>101</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Cache Size :</string>
       </property>
      </widget>
      <widget class="QLabel" name="model_name">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>40</y>
         <width>301</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="cpu_mhz">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>70</y>
         <width>281</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="cpu_cores">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>100</y>
         <width>281</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="cache_size">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>130</y>
         <width>281</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>210</y>
        <width>461</width>
        <height>131</height>
       </rect>
      </property>
      <property name="title">
       <string>内存基本信息</string>
      </property>
      <widget class="QProgressBar" name="memory_usage">
       <property name="geometry">
        <rect>
         <x>160</x>
         <y>40</y>
         <width>251</width>
         <height>21</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>40</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Memory</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_16">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>70</y>
         <width>41</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Total :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_17">
       <property name="geometry">
        <rect>
         <x>180</x>
         <y>100</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Free :</string>
       </property>
      </widget>
      <widget class="QLabel" name="mem_total">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>70</y>
         <width>121</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="mem_free">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>100</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="MODULE_info">
     <attribute name="title">
      <string>模块信息</string>
     </attribute>
     <widget class="QTableWidget" name="module_table">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>421</width>
        <height>371</height>
       </rect>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="alternatingRowColors">
       <bool>false</bool>
      </property>
      <property name="showGrid">
       <bool>false</bool>
      </property>
      <column>
       <property name="text">
        <string>Module Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Module Size</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Instances Num</string>
       </property>
      </column>
     </widget>
     <widget class="QPushButton" name="pushButton">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>340</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Install module</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>10</y>
        <width>361</width>
        <height>311</height>
       </rect>
      </property>
      <property name="title">
       <string>Module Information</string>
      </property>
      <widget class="QPushButton" name="uninstall_module">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>270</y>
         <width>131</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>Uninstall module</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_21">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Module Name :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_22">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>80</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Module Size :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_23">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Instances Num :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_27">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>220</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Module State :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_28">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>121</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Dependent Mod:</string>
       </property>
      </widget>
      <widget class="QLabel" name="module_name">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>40</y>
         <width>211</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="module_size">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>80</y>
         <width>211</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="ins_num">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>120</y>
         <width>211</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="dependent">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>160</y>
         <width>211</width>
         <height>21</height>
        </rect>
       </property>
       <property name="sizeIncrement">
        <size>
         <width>0</width>
         <height>2</height>
        </size>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="module_state">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>220</y>
         <width>151</width>
         <height>21</height>
        </rect>
       </property>
       <property name="sizeIncrement">
        <size>
         <width>0</width>
         <height>2</height>
        </size>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="SYS_info">
     <attribute name="title">
      <string>系统信息</string>
     </attribute>
     <widget class="QGroupBox" name="system_info">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>30</y>
        <width>491</width>
        <height>161</height>
       </rect>
      </property>
      <property name="title">
       <string>System Information</string>
      </property>
      <widget class="QLabel" name="label_18">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>kernel version :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_19">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>70</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>gcc version :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_20">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>110</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>build date :</string>
       </property>
      </widget>
      <widget class="QLabel" name="kernel_version">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>30</y>
         <width>311</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="gcc_version">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>70</y>
         <width>341</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
      <widget class="QLabel" name="build_date">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>110</y>
         <width>321</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="About">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>210</y>
        <width>401</width>
        <height>161</height>
       </rect>
      </property>
      <property name="title">
       <string>About Me</string>
      </property>
      <widget class="QLabel" name="label_24">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>111</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>Author :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_25">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>70</y>
         <width>111</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>e-mail :</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_26">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>110</y>
         <width>101</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Student ID :</string>
       </property>
      </widget>
      <widget class="QLabel" name="kernel_version_3">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>30</y>
         <width>201</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>Leslie Lee</string>
       </property>
      </widget>
      <widget class="QLabel" name="email">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>70</y>
         <width>201</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>chenchenlcl@outlook.com</string>
       </property>
      </widget>
      <widget class="QLabel" name="ID">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>110</y>
         <width>201</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>U201714863</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>869</width>
     <height>28</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuLeslie_s_OS_Cource_Design">
    <property name="title">
     <string>Leslie's OS Cource Design</string>
    </property>
   </widget>
   <addaction name="menuLeslie_s_OS_Cource_Design"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
